src/SoundOptionsModal.test.js

```diff
@@ -0,0 +1,154 @@
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(true);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(true);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(true);
+    });
+    test('Changing audioEnabled, announcementsEnabled and sonificationEnabled properties to false should turn off corresponding toggle siwtches', () => {
+        expect.assertions(8);
+        const { wrapper } = createMountSoundOptionsModal({ audioEnabled: false, announcementsEnabled: false, sonificationEnabled: false });
+        // Other options than allSounds should have disabled class when audioEnabled is false
+        expect(wrapper.find('.SoundOptions__option--disabled').length).toBe(2);
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(false);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(false);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(false);
+    })
+});
+
+describe('When interacting with footer buttons', () => {
+    test('Pressing cancel button should close the modal without changing toggle states', () => {
+        expect.assertions(10);
+        const { wrapper, mockOnCancel } = createMountSoundOptionsModal();
+        expect(getWrapperState(wrapper).audioEnabled).toBe(true);
+        expect(getWrapperState(wrapper).announcementsEnabled).toBe(true);
+        expect(getWrapperState(wrapper).sonificationEnabled).toBe(true);
+
+        const announcementsToggle = getAudioToggleSwitches(wrapper).at(1);
+        const sonificationToggle = getAudioToggleSwitches(wrapper).at(2);
+        announcementsToggle.simulate('click');
+        sonificationToggle.simulate('click');
+        expect(getWrapperState(wrapper).announcementsEnabled).toBe(false);
+        expect(getWrapperState(wrapper).sonificationEnabled).toBe(false);
+
+        const allSoundsToggle = getAudioToggleSwitches(wrapper).at(0);
+        allSoundsToggle.simulate('click');
+        expect(getWrapperState(wrapper).audioEnabled).toBe(false);
+
+        const cancelButton = getCancelButton(wrapper);
+        cancelButton.at(0).simulate('click');
+        expect(mockOnCancel.mock.calls.length).toBe(1);
+        // Reset all toggle state to the initial state from props
```

Comment:
I'll leave a comment also on `SoundOptionsModal.js`, but this isn't quite what we want. The `SoundOptionsModal` attempts to keep its state consistent with that provided in the props by setting them at cancel. However, it is possible to change the announcement setting using a keyboard shortcut, and when this is done, the dialog will show the wrong value. To reproduce:

1. Load Weavly
2. Turn on keyboard shortcuts
2. Verify that announcements are enabled by opening the Delete All dialog
3. Use the keyboard shortcut to disable the announcements (Ctrl + Alt + x, x)
4. Verify that announcements are disabled by opening the Delete All dialog
5. Open the Sound Options dialog

Expected:

* The announcements toggle switch to be off

Actual:

* The announcements toggle switch is on

The fix is to do like in the `WorldSelector` and set the state of the Sound Options dialog when it is shown: https://github.com/codelearncreate/c2lc-coding-environment/blob/818ac96dbffc1b98e08ec0dd8a9b23a82402beb9/src/WorldSelector.js#L138-L144

Reply:
And if we want to change toggling of the announcements while the dialog is open, then we would need to check for that also.